frmts/gtiff/cogdriver.cpp

```diff
@@ -629,6 +629,13 @@ static std::unique_ptr<GDALDataset> CreateReprojectedDS(
         papszArg = CSLAddString(papszArg, "-wo");
         papszArg = CSLAddString(
             papszArg, (CPLString("NUM_THREADS=") + pszNumThreads).c_str());
+
+        const char *pszWarpThreads =
+            CPLGetConfigOption("GDAL_NUM_THREADS", nullptr);
+        if (!pszWarpThreads)
+        {
+            CPLSetConfigOption("GDAL_NUM_THREADS", pszNumThreads);
```

Comment:
Thanks for the feedback, I will switch it across.

Reply:
as my C/C++ knowledge is super rusty, I think I have done what you have suggested.

I have converted it to a `std::unique_ptr` which is only assigned if `pszNumThreads` is set.

Having this piece of code seems to work in my test cases reducing the COG creation time almost 50-75%.

Reply:
Since I am now doing the comparison twice, granted its just a `if (pszNum)` which should be super quick, does it make more sense to move this logic back into the if statement with the rest of the `pszNumThreads` logic on 627?

Reply:
The code looks fine as it is. Merging